moml/data/processors/process_chemical_data.py

```
@@ -0,0 +1,217 @@
+        logger.error("SMILES column missing in dataframe, cannot engineer features")
+        raise KeyError("SMILES column missing in dataframe")
+
+    # Add a column to cache parsed molecules to avoid redundant parsing
+    if "rdkit_mol_cache" not in df.columns:
+        # Safely validate SMILES entries and create molecule cache
+        def safe_parse_smiles(s):
+            if pd.isna(s) or not isinstance(s, str):
+                return None
+            try:
+                return Chem.MolFromSmiles(str(s))
+            except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)